static/app/components/tabs/tab.tsx

```diff
@@ -99,13 +99,28 @@ export const BaseTab = forwardRef(
           borderStyle={borderStyle}
           ref={ref}
         >
-          <FilledStyledInteractionStateLayer hasSelectedBackground={false} />
-          <FilledFocusLayer />
+          <VariantStyledInteractionStateLayer hasSelectedBackground={false} />
+          <VariantFocusLayer />
           {props.children}
         </FilledTabWrap>
       );
     }
 
+    if (variant === 'floating') {
+      return (
+        <FloatingTabWrap
+          {...tabProps}
+          hidden={hidden}
+          overflowing={overflowing}
+          ref={ref}
+        >
+          <VariantStyledInteractionStateLayer />
```

Comment:
One thing I'm not seeing are pressed styles (when you hold the mouse down) and hover styles on the selected tab. I think you can fix that by adding `hasSelectedBackground={false}` and adding a background in the CSS when it's selected.